test/acceptance/helpers/helpers.go

```diff
@@ -58,6 +63,55 @@ func WaitForAllPodsToBeReady(t *testing.T, client kubernetes.Interface, namespac
 	})
 }
 
+// Deploy creates a Kubernetes deployment by applying configuration stored at filepath,
+// sets up a cleanup function and waits for the deployment to become available.
+func Deploy(t *testing.T, options *k8s.KubectlOptions, noCleanupOnFailure bool, filepath string) {
+	t.Helper()
+
+	KubectlApply(t, options, filepath)
+
+	file, err := os.Open(filepath)
+	require.NoError(t, err)
+
+	deployment := v1.Deployment{}
+	err = yaml.NewYAMLOrJSONDecoder(file, 1024).Decode(&deployment)
+	require.NoError(t, err)
+
+	Cleanup(t, noCleanupOnFailure, func() {
+		// Note: this delete command won't wait for pods to be fully terminated.
+		// This shouldn't cause any test pollution because the underlying
+		// objects are deployments, and so when other tests create these
+		// they should have different pod names.
+		KubectlDelete(t, options, filepath)
+	})
+
+	RunKubectl(t, options, "wait", "--for=condition=available", fmt.Sprintf("deploy/%s", deployment.Name))
+}
+
+// checkConnection execs into a pod of the deployment given by deploymentName
+// and runs a curl command with the provided curlArgs.
+// If expectSuccess is true, it will expect connection to succeed,
+// otherwise it will expect failure due to intentions.
+func CheckConnection(t *testing.T, options *k8s.KubectlOptions, deploymentName string, expectSuccess bool, curlArgs ...string) {
+	t.Helper()
+
+	retrier := &retry.Timer{Timeout: 20 * time.Second, Wait: 500 * time.Millisecond}
+
+	args := []string{"exec", "deploy/" + deploymentName, "-c", deploymentName, "--", "curl", "-vvvsSf"}
+	args = append(args, curlArgs...)
+
+	retry.RunWith(retrier, t, func(r *retry.R) {
+		output, err := RunKubectlAndGetOutputE(t, options, args...)
+		if expectSuccess {
+			require.NoError(r, err)
+			require.Contains(r, output, "hello world")
```

Comment:
Great work pulling this one in!
Can we modify this to pass in some form of expected output?
"hello world" string is not going to always be in every test, I know that in the deployment used by terminating gateways it looks for a different string.

Also I wonder if the output for failure cases will vary based on the type of test?

Reply:
We could also call this `CheckStaticServerConnection`. Then it's clear that we're talking to the static-server and its safe to assume its output.

Reply:
Yeah, let me change the name. So far, all tests are using the static server and so the output is the same. We can make this more generic later.

Reply:
Renamed! Good catch!